ckanext/datapusher_plus/migration/datapusher_plus/versions/01_e9c4a88839c8_upgrade_jobs_table.py

```diff
@@ -5,37 +5,44 @@
 Create Date: 2023-09-22 22:14:35.137116
 
 """
+
 from alembic import op
 import sqlalchemy as sa
 
 
 # revision identifiers, used by Alembic.
-revision = 'e9c4a88839c8'
+revision = "e9c4a88839c8"
 down_revision = None
 branch_labels = None
 depends_on = None
 
 
 def upgrade():
-    #upgrade jobs table if it not exists
-    
-    op.add_column(
-        u'jobs',
-        sa.Column(
-            'aps_job_id',
-            sa.UnicodeText),
-    )
-    
-    #upgrade logs table
-    op.add_column(
-        'logs',
-        sa.Column(
-            'id',
-            sa.Integer,
-            primary_key=True,
-            autoincrement=True),
-    )
-
-    
+    # upgrade jobs table if it not exists
+    if _check_column_exists("jobs", "aps_job_id"):
+        return
+    else:
+        op.add_column(
+            "jobs",
+            sa.Column("aps_job_id", sa.UnicodeText),
+        )
+
+    # upgrade logs table
+    if _check_column_exists("logs", "id"):
+        return
+    else:
+        op.add_column(
+            "logs",
+            sa.Column("id", sa.Integer, primary_key=True, autoincrement=True),
+        )
```

Comment:
@tino097 I think it is better to not return. since the existance of "aps_job_id" will cause the logic of "logs" to never be executed.

I changed it to the following locally and I was able to evercome the `jobs.id` column does not exist error.
```suggestion
    # upgrade jobs table if it not exists
    if not _check_column_exists("jobs", "aps_job_id"):
        op.add_column(
            "jobs",
            sa.Column("aps_job_id", sa.UnicodeText),
        )

    # upgrade logs table
    if not _check_column_exists("logs", "id"):
        op.add_column(
            "logs",
            sa.Column("id", sa.Integer, primary_key=True, autoincrement=True),
        )
```

Reply:
Thanks for this, i will try to cherry pick some of the commits from this branch and merge those into `dev-1.0` as i think this branch will get discarded